skill/response.py

```diff
@@ -0,0 +1,163 @@
+
+        Args:
+            utterance: The words that comprise the user's request.
+        """
+        self.requested_location = find_in_utterance(utterance, self.location_regex_path)
+        self._get_geolocation()
+        self.date_time = self._get_current_datetime()
+        self.speak_am_pm = not self.time_format_24_hour and self.geolocation is not None
+        self.dialog_data = dict(time=self._get_speakable_time())
+        if self.geolocation is None:
+            self.dialog_name = "time-current-local"
+        else:
+            self.dialog_name = "time-current-location"
+            self.dialog_data.update(location=self.get_display_location())
+
+    def build_future_time_response(self, utterance):
+        """Build a response to a request for a time in the future.
+
+        Time can be requested locally or in a specific city.
+
+        Args:
+            utterance: The words that comprise the user's request.
+        """
+        utterance = normalize(utterance.lower())
+        current_time_offset = get_duration_from_utterance(utterance)
+        self.date_time, remaining_utterance = get_datetime_from_utterance(utterance)
+        if self.date_time is not None:
+            self.requested_location = find_in_utterance(
+                remaining_utterance, self.location_regex_path
+            )
+            self._get_geolocation()
+            if self.geolocation is not None:
+                tz_info = pytz.timezone(self.geolocation["timezone"])
+                self.date_time = self.date_time.astimezone(tz_info)
+            self.speak_am_pm = True
+            self.dialog_data = dict(
+                time=self._get_speakable_time(),
+                duration=nice_duration(current_time_offset),
+            )
+            if self.geolocation is None:
+                self.dialog_name = "time-future-local"
+            else:
+                self.dialog_name = "time-future-location"
+                self.dialog_data.update(location=self.get_display_location())
+
+    # TODO: abstract out into a skill library
+    def _get_geolocation(self):
+        """Retrieve geographical data for a city from the Selene Geolocation API."""
+        if self.requested_location is None:
+            LOG.info("No location found in request")
+        else:
+            LOG.info("Location found in request: " + self.requested_location)
+            self.geolocation = get_geolocation(self.requested_location)
+
+    def _get_speakable_time(self) -> str:
+        """Convert a datetime object into a string that can be passed to TTS.
+
+        Returns:
+            The words to be spoken in response to the time request.
+        """
+        speakable_time = nice_time(
+            self.date_time,
+            speech=True,
+            use_24hour=self.time_format_24_hour,
+            use_ampm=self.speak_am_pm,
+        )
+        # HACK: Mimic 2 has a bug with saying "AM".  Work around it for now.
```

Comment:
If mimic is supposed to say Aaa Mmm when given AM, how does it say the word am, as in I am?

Reply:
Somebody certainly thought so.  I copied this comment from the date/time skill.  The correct thing to do here is probably to change Lingua Franca to add the periods after the letters. But I do not know this for sure.